### PR TITLE
Switch data folder during bead runs

### PR DESCRIPTION
**Pull request by figuernd**

### CODE REVIEW

File: configs/example.yaml [hunk 1]
```diff
@@ -64,6 +64,7 @@ ifcb:
     port: 8092
     serial: "111-111-111"
     routines_dir: "/routines"
+    data_dir: "/mnt/data/ifcbdata"
```
Comment (rgov) [inline configs/example.yaml:67]: I think it's worth a comment:

that this path is interpreted by IFCBacquire, which may be in a separate container.


```yaml
    # This path is interpreted by IFCBacquire, which may be in another
    # container or namespace.
    data_dir: "/mnt/data/ifcbdata"
```

File: src/ifcb/src/ifcb/node.py [hunk 2]
```diff
@@ -203,6 +212,10 @@ def main():
     client.on(('triggerrois',),
               functools.partial(on_triggerrois, roi_pub, mkr_pub))
 
+    # Set up callbacks for datafolder switching
+    client.on(('valuechanged','interactive','stopped',),
```
Comment (rgov) [inline src/ifcb/src/ifcb/node.py:216]: FYI: The terminal comma here isn't needed, that's only for a single entry. `(a,)` is a tuple of one element, `(a,b)` is a tuple of two. No fix necessary.

File: src/ifcb/src/ifcb/node.py [hunk 3]
```diff
@@ -49,6 +48,13 @@ def do_runroutine(client, pub, req):
     if '/' in req.routine:
         return srv.RunRoutineResponse(success=False)
 
+    # Check for beads routines
+    assert rospy.get_param("~data_dir") is not None
```
Comment (rgov) [inline src/ifcb/src/ifcb/node.py:52]: > KeyError is raised if the parameter is not set.

I don't think this assertion works anymore. I would remove this line.

**Review by rgov — CHANGES_REQUESTED** [review]

File: src/ifcb/src/ifcb/node.py (hunk 2 above)
Comment (figuernd) [reply src/ifcb/src/ifcb/node.py:215]: Ahh that makes more sense, thanks.

File: configs/example.yaml (hunk 1 above)
Comment (figuernd) [reply configs/example.yaml:69]: Done

**Pull request merged**